File: SERVER-HTTP/src/main/scala/com/variant/server/boot/ServerBoot.scala
package com.variant.server.boot

import com.variant.core.impl.VariantComptime
import com.variant.core.util.VariantStringUtils
import com.variant.core.impl.VariantCore
import net.liftweb.common.Loggable
import com.variant.server.ServerProperties


/**
 * Entry point into the Variant Server. 
 *
object ServerBootOld extends Loggable {
  
   var core: VariantCore = null
   lazy val properties : ServerProperties = new ServerProperties(core)
   
   val bootTime = System.currentTimeMillis()
   
   **
    * External configuration passes the name of the resource properties file here.
    *
   def init(configNamesAsResources: String*) : Unit = {
      core = new VariantCore(configNamesAsResources:_*)
      core.getComptime().registerComponent(VariantComptime.Component.SERVER, "0.6.1")

		if (logger.isDebugEnabled) {
			LOG.debug("+-- Bootstrapping Variant Server with following application properties: --");
			for (key <- properties) {
				logger.debug("| " + key.propName() + " = " + properties.get(key, java.lang.String) + " : " + properties.getSource(key));
			}
			LOG.debug("+------------- Fingers crossed, this is not PRODUCTION -------------");
		}

   }
}
*/

File: SERVER/app/com/variant/server/ServerBoot.java
/*
package com.variant.server;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.variant.core.VariantCorePropertyKeys.Key;
import com.variant.core.VariantProperties;
import com.variant.core.impl.VariantComptime;
import com.variant.core.impl.VariantCore;

public class ServerBoot {

	public static final long bootTime = System.currentTimeMillis();

	private static final Logger LOG = LoggerFactory.getLogger(ServerBoot.class);
	private static VariantCore core = null;
			   
			   
	/**
	 * No argument because tests will boot up lift and therefore call this the same way as main.
	 * Test overrides have to be done via variant.props
	 *
	public static void boot()  {
		core = new VariantCore();
		core.getComptime().registerComponent(VariantComptime.Component.SERVER, "0.6.3");
		
		if (LOG.isDebugEnabled()) {
			VariantProperties props = core.getProperties();
			// Touch ServerProperties class to cause it's instantiation and hence registration of keys with superclass.
			//ServerProperties.SESSION_STORE_VACUUM_INTERVAL_SECS;
			LOG.debug("+-- Bootstrapping Variant Server with following application properties: --");
			for (Key key: Key.keys(ServerPropertyKeys.class)) {
				LOG.debug("| " + key.propertyName() + " = " + props.get(key, String.class) + " : " + props.getSource(key));
			}
			LOG.debug("+------------- Fingers crossed, this is not PRODUCTION -------------");
		}
	}
		
	/**
	 * 
	 * @return
	 *
	public static VariantCore getCore() {
		return core;
	}
}
*/